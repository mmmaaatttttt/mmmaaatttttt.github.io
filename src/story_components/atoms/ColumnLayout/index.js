import styled, { css } from "styled-components";
import media from "utils/media";

const ColumnLayout = styled.div`
  display: flex;

  & > div {
    flex: 1;
  }

  ${props =>
    props.break &&
    css`
      ${media[props.break]`
      flex-direction: column;
    `};
    `};
`;

export default ColumnLayout;
