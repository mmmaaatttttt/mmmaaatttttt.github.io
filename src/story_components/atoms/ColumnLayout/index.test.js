import React from "react";
import { render } from "@testing-library/react";
import ColumnLayout from "./";

it("renders without crashing", () => {
  render(<ColumnLayout />);
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <ColumnLayout break="large">
      <div>Here's a child</div>
      <div>Here's another</div>
    </ColumnLayout>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("lets you make columns of different widths", () => {
  const { asFragment } = render(
    <ColumnLayout break="small" sizes={[3, 2]}>
      <div>Here's a child</div>
      <div>Here's another</div>
    </ColumnLayout>
  );
  expect(asFragment()).toMatchSnapshot();
});
