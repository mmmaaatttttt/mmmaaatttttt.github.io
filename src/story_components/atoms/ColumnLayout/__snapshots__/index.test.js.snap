// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lets you make columns of different widths 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa hWapkf"
    sizes="3,2"
  >
    <div>
      Here's a child
    </div>
    <div>
      Here's another
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa hgMGHa"
  >
    <div>
      Here's a child
    </div>
    <div>
      Here's another
    </div>
  </div>
</DocumentFragment>
`;
